src/Servers/Kestrel/Core/test/PathNormalizerTests.cs

```diff
@@ -54,11 +54,24 @@ public class PathNormalizerTests
     [InlineData("/", "/")]
     [InlineData("/no/segments", "/no/segments")]
     [InlineData("/no/segments/", "/no/segments/")]
+    [InlineData("/././", "/")]
+    [InlineData("/./.", "/")]
+    [InlineData("/../..", "/")]
+    [InlineData("/../../", "/")]
+    [InlineData("/../.", "/")]
+    [InlineData("/./..", "/")]
+    [InlineData("/.././", "/")]
+    [InlineData("/./../", "/")]
+    [InlineData("/..", "/")]
+    [InlineData("/.", "/")]
+    [InlineData("/a/abc/../abc/../b", "/a/b")]
+    [InlineData("/a/abc/.a", "/a/abc/.a")]
```

Comment:
Should there be a test like `/a/.b/c`? I don't see any where a `.X` is in the middle of the path.

Reply:
I will add a test for it.

Reply:
I will add a few more on top of this.